src/renderer/gdi/paint.cpp

```diff
@@ -419,7 +419,7 @@ bool GdiEngine::FontHasWesternScript(HDC hdc)
         pPolyTextLine->rcl.top = pPolyTextLine->y + topOffset;
         pPolyTextLine->rcl.right = pPolyTextLine->rcl.left + (til::CoordType)cchCharWidths;
         pPolyTextLine->rcl.bottom = pPolyTextLine->y + coordFontSize.height - bottomOffset;
-        pPolyTextLine->pdx = polyWidth.data();
+        pPolyTextLine->pdx = reinterpret_cast<int*>(polyWidth.data());

```

Comment:
Cast char32_t* to int*, Win32 API requires this.

Reply:
As you probably know, `int` is not guaranteed to be 32 bit large. I think what we should try to do instead is simply use a vector:
```cpp
std::pmr::vector<std::pmr::vector<int>> _polyWidths;
```

Since the small vector optimization in the Microsoft STL is fairly bad anyway, I don't think there's any benefit in using it here. Any non-trivial amount of text will immediately exceed the small-vector size in a `std::basic_string`. It's also not beneficial given that we use a PMR pool too.